llmfoundry/data/denoising.py

```diff
@@ -476,7 +476,7 @@ def build_text_denoising_dataloader(
         download_timeout=cfg.dataset.get('download_timeout', 60),
         validate_hash=cfg.dataset.get('validate_hash'),
         shuffle_seed=cfg.dataset.get('shuffle_seed'),
-        num_canonical_nodes=cfg.dataset.get('num_canonical_nodes'),
+        num_canonical_nodes=cfg.dataset.get('num_canonical_nodes', 128),
```

Comment:
y not just allows this to be `None`?
`None` is the [default in streaming](https://docs.mosaicml.com/projects/streaming/en/stable/api_reference/generated/streaming.StreamingDataset.html)

Reply:
Not safe for elastic determinism, there is an ongoing discussion in Slack but I really want this repo to be deterministic by default.